src/Treeview.ts

```diff
@@ -262,42 +266,40 @@ export class JsonOutlineProvider implements vscode.TreeDataProvider<string> {
     }
 
     // tslint:disable-next-line:cyclomatic-complexity // Grandfathered in
-    private getIconPath(elementInfo: IElementInfo): string {
+    private getIconPath(elementInfo: IElementInfo): string | undefined {
 
         let icon: string;
-        const keyNode = this.tree.getValueAtCharacterIndex(elementInfo.current.key.start);
+        const keyOrResourceNode = this.tree.getValueAtCharacterIndex(elementInfo.current.key.start);
 
         // Is current element a root element?
         if (elementInfo.current.level === 1) {
-            if (keyNode._value.toUpperCase() === "$schema".toUpperCase()) { icon = "label.svg" }
-            if (keyNode._value.toUpperCase() === "version".toUpperCase()) { icon = "label.svg" }
-            if (keyNode._value.toUpperCase() === "contentVersion".toUpperCase()) { icon = "label.svg" }
-            if (keyNode._value.toUpperCase() === "handler".toUpperCase()) { icon = "label.svg" }
-            if (keyNode._value.toUpperCase() === "parameters".toUpperCase()) { icon = "parameters.svg" }
-            if (keyNode._value.toUpperCase() === "variables".toUpperCase()) { icon = "variables.svg" }
-            if (keyNode._value.toUpperCase() === "resources".toUpperCase()) { icon = "resources.svg" }
-            if (keyNode._value.toUpperCase() === "outputs".toUpperCase()) { icon = "outputs.svg" }
-
+            if (keyOrResourceNode.toString().toUpperCase() === "$schema".toUpperCase()) { icon = "label.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "version".toUpperCase()) { icon = "label.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "contentVersion".toUpperCase()) { icon = "label.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "handler".toUpperCase()) { icon = "label.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "parameters".toUpperCase()) { icon = "parameters.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "variables".toUpperCase()) { icon = "variables.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "resources".toUpperCase()) { icon = "resources.svg" }
+            if (keyOrResourceNode.toString().toUpperCase() === "outputs".toUpperCase()) { icon = "outputs.svg" }
```

Comment:
Why does the right hand expression in each condition have a toUppercase()?

Reply:
NMC